api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxCurrCohortQueries.java

```diff
@@ -120,54 +149,141 @@ public CohortDefinition getTxCurrCompositionCohort(String cohortName, boolean cu
     txCurrComposition
         .getSearches()
         .put(
-            "8",
+            "7",
+            EptsReportUtils.map(
+                getDeadPatientsInDemographiscByReportingEndDate(),
+                "onOrBefore=${onOrBefore},location=${location}"));
+    txCurrComposition
+        .getSearches()
+        .put(
+            "888",
             EptsReportUtils.map(
                 getPatientsReportedAsAbandonmentButStillInPeriod(),
                 String.format(
                     "onOrBefore=${onOrBefore},location=${location},abandonmentDays=%s",
                     abandonmentDays)));
+    txCurrComposition
+        .getSearches()
+        .put(
+            "8",
+            EptsReportUtils.map(
+                getPatientDeathRegisteredInLastHomeVisitCardByReportingEndDate(),
+                "onOrBefore=${onOrBefore},location=${location}"));
+    txCurrComposition
+        .getSearches()
+        .put(
+            "9",
+            EptsReportUtils.map(
+                getDeadPatientsInFichaResumeAndClinicaOfMasterCardByReportEndDate(),
+                "onOrBefore=${onOrBefore},location=${location}"));
+    txCurrComposition
+        .getSearches()
+        .put(
+            "10",
+            EptsReportUtils.map(
+                getTransferredOutPatientsInFichaResumeAndClinicaOfMasterCardByReportEndDate(),
+                "onOrBefore=${onOrBefore},location=${location}"));
     txCurrComposition
         .getSearches()
         .put(
             "11",
             EptsReportUtils.map(
-                getPatientsWithNextPickupDate(), "onOrBefore=${onOrBefore},location=${location}"));
+                getPatientSuspendedInFichaResumeAndClinicaOfMasterCardByReportEndDate(),
+                "onOrBefore=${onOrBefore},location=${location}"));
     txCurrComposition
         .getSearches()
         .put(
             "12",
             EptsReportUtils.map(
-                getPatientsWithNextConsultationDate(),
+                getPatientWhoAfterMostRecentDateHaveDrusPickupOrConsultation(),
+                "location=${location}"));
+    // section 3
+    txCurrComposition
+        .getSearches()
+        .put(
+            "13",
+            EptsReportUtils.map(
+                getPatientHavingLastScheduledDrugPickupDate(),
                 "onOrBefore=${onOrBefore},location=${location}"));
+    txCurrComposition
+        .getSearches()
+        .put(
+            "14",
+            EptsReportUtils.map(
+                getPatientWithoutScheduledDrugPickupDateMasterCardAmdArtPickup(),
+                "location=${location}"));
 
     String compositionString;
     if (currentSpec) {
       compositionString =
-          "(1 OR 2 OR 3 OR 4) AND (NOT (5 OR ((6 OR (NOT 11)) AND (NOT (7 OR 8))))) AND (11 OR 12)";
+          "(1 OR 2 OR 3 OR 4 OR 5) AND NOT ((6 OR 7 OR 8 OR 9 OR 10 OR 11) AND NOT 12) AND NOT (13 OR 14)";
```

Comment:
I think we need change the brackets in the composition string here.

Pinki left a note that the order of hte exclusions is important, i.e. we need to exclude the patients who left ART before then excluding the LTFU patients.

So soemthing like: `(main query AND NOT patient who left) NOT LTFU...etc...`